runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -1805,6 +1805,40 @@ TR_J9ServerVM::getInvokeExactThunkHelperAddress(TR::Compilation *comp, TR::Symbo
    return helper;
    }
 
+UDATA
+TR_J9ServerVM::getCellSizeForSizeClass(uintptr_t sizeClass)
+   {
+#if defined(J9VM_GC_REALTIME)
+   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
+   stream->write(JITServer::MessageType::VM_getCellSizeForSizeClass, sizeClass);
+   return std::get<0>(stream->read<UDATA>());
+#endif
+   return 0;
+   }
+
+UDATA
+TR_J9ServerVM::getObjectSizeClass(uintptr_t objectSize)
+   {
+#if defined(J9VM_GC_REALTIME)
+   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
+   stream->write(JITServer::MessageType::VM_getObjectSizeClass, objectSize);
+   return std::get<0>(stream->read<UDATA>());
+#endif
+   return 0;
+   }
+
+// Creates a node to initialize the local object flags field
+//
+TR::Node *
+TR_J9ServerVM::initializeLocalObjectFlags(TR::Compilation * comp, TR::Node * allocationNode, TR_OpaqueClassBlock * ramClass)
+   {
+   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
+   stream->write(JITServer::MessageType::VM_initializeLocalObjectFlags, ramClass);
```

Comment:
I think this will result in many messages being sent.
The original code reads some value from the compilation thread (which could be any) to be used at runtime by any other application thread. This makes me think that all the threads have the same value which does not change in time. We need to verify that, but if that is the case then we can fetch that specific value once during the VMinit message.

Reply:
> I think this will result in many messages being sent.

So far the local sanity test shows that it doesn't look like many `VM_initializeLocalObjectFlags ` messages are sent for the following tests. I extracted out `VM_getSystemClassFromClassName`, `VM_getClassFromSignature`, 
`VM_getJ2IThunk`, and `VM_initializeLocalObjectFlags` as comparison. 

```
Running test jit_jitt_0 ...
#0208       3 VM_getSystemClassFromClassName
#0217    8679 VM_getClassFromSignature
#0307    1059 VM_getJ2IThunk
#0312       6 VM_initializeLocalObjectFlags

Running test jit_jitt_1 ...
#0208       3 VM_getSystemClassFromClassName
#0217    8697 VM_getClassFromSignature
#0307    1059 VM_getJ2IThunk
#0312       6 VM_initializeLocalObjectFlags

Running test jit_jitt_2 ...
#0208       3 VM_getSystemClassFromClassName
#0217    8702 VM_getClassFromSignature
#0307    1059 VM_getJ2IThunk
#0312       6 VM_initializeLocalObjectFlags

Running test jit_jitt_3 ...
#0208       3 VM_getSystemClassFromClassName
#0217    8696 VM_getClassFromSignature
#0307    1059 VM_getJ2IThunk
#0312       6 VM_initializeLocalObjectFlags

Running test jit_compareAndBranch_0 ...
#0208       2 VM_getSystemClassFromClassName
#0217       4 VM_getClassFromSignature

Running test jit_jitt_openj9_none_SCC_0 ...
#0208       3 VM_getSystemClassFromClassName
#0217    8704 VM_getClassFromSignature
#0307    1059 VM_getJ2IThunk
#0312       6 VM_initializeLocalObjectFlags

Running test jit_jitt_openj9_none_SCC_1 ...
#0208       3 VM_getSystemClassFromClassName
#0217    8703 VM_getClassFromSignature
#0307    1059 VM_getJ2IThunk
#0312       6 VM_initializeLocalObjectFlags

Running test jit_jitt_openj9_none_SCC_2 ...
#0208       3 VM_getSystemClassFromClassName
#0217    8698 VM_getClassFromSignature
#0307    1059 VM_getJ2IThunk
#0312       6 VM_initializeLocalObjectFlags

Running test jit_jitt_openj9_none_SCC_3 ...
#0208       3 VM_getSystemClassFromClassName
#0217    8679 VM_getClassFromSignature
#0307    1059 VM_getJ2IThunk
#0312       6 VM_initializeLocalObjectFlags
```

Reply:
The local sanity functional test shows that the number of `VM_initializeLocalObjectFlags` messages is relatively low between `1~16`, except for`StringPeepholeTest_0` which exchanges `~141` messages. Wondering if message `VM_initializeLocalObjectFlags` should be kept for now to keep it functionally correct on the realTimeGC path.

```
Type# #called TypeName
#0000    5427 compilationCode
#0002    6711 mirrorResolvedJ9Method
#0003    4731 get_params_to_construct_TR_j9method
#0004       1 getUnloadedClassRanges
#0105    4730 ResolvedMethod_staticAttributes
#0106    6483 ResolvedMethod_getClassFromConstantPool
#0107    5634 ResolvedMethod_getDeclaringClassFromFieldOrStatic
#0108    3239 ResolvedMethod_classOfStatic
#0109     200 ResolvedMethod_startAddressForJNIMethod
#0110   18954 ResolvedMethod_fieldAttributes
#0111   10629 ResolvedMethod_getResolvedStaticMethodAndMirror
#0112   15537 ResolvedMethod_getResolvedSpecialMethodAndMirror
#0113     153 ResolvedMethod_classCPIndexOfMethod
#0116   18218 ResolvedMethod_getResolvedPossiblyPrivateVirtualMethodAndMirror
#0118   15544 ResolvedMethod_getResolvedInterfaceMethod_2
#0119     209 ResolvedMethod_getResolvedInterfaceMethodAndMirror_3
#0120     305 ResolvedMethod_getResolvedInterfaceMethodOffset
#0125    9934 ResolvedMethod_getRemoteROMString
#0126    1240 ResolvedMethod_fieldOrStaticName
#0127     662 ResolvedMethod_getRemoteROMClassAndMethods
#0128       5 ResolvedMethod_getResolvedHandleMethod
#0129       5 ResolvedMethod_isUnresolvedMethodTypeTableEntry
#0130       5 ResolvedMethod_methodTypeTableEntryAddress
#0145      34 ResolvedMethod_setClassForNewInstance
#0147    5235 ResolvedMethod_getResolvedImproperInterfaceMethodAndMirror
#0149   10921 ResolvedMethod_stringConstant
#0150     833 ResolvedMethod_getResolvedVirtualMethod
#0151    2418 ResolvedMethod_getMultipleResolvedMethods
#0157    3828 ResolvedMethod_definingClassFromCPFieldRef
#0208       2 VM_getSystemClassFromClassName
#0211       1 VM_getClassClassPointer
#0212      84 VM_setJ2IThunk
#0217   26552 VM_getClassFromSignature
#0218     241 VM_jitFieldsAreSame
#0219    1117 VM_jitStaticsAreSame
#0222      50 VM_classHasBeenExtended
#0229     272 VM_isClassInitialized
#0231      48 VM_getInitialLockword
#0232     562 VM_isString1
#0239     133 VM_stackWalkerMaySkipFrames
#0245    3202 VM_getClassFromNewArrayType
#0248     682 VM_getArrayClassFromComponentClass
#0258     142 VM_getOffsetOfClassFromJavaLangClassField
#0270      20 VM_getInstanceFieldOffset
#0273     575 VM_getClassDepthAndFlagsValue
#0274     798 VM_getMethodFromName
#0278    1046 VM_getReferenceSlotsInClass
#0282      23 VM_getInterpreterVTableSlot
#0291      18 VM_getObjectNewInstanceImplMethod
#0294       5 VM_setInvokeExactJ2IThunk
#0295       2 VM_createMethodHandleArchetypeSpecimen
#0297      21 VM_instanceOfOrCheckCast
#0300      77 VM_getResolvedMethodsAndMirror
#0301       1 VM_getVMInfo
#0303      97 VM_dereferenceStaticAddress
#0307     857 VM_getJ2IThunk
#0308       5 VM_needsInvokeExactJ2IThunk
#0312     141 VM_initializeLocalObjectFlags
#0400    4339 CompInfo_isCompiled
#0401      18 CompInfo_getInvocationCount
#0408      18 CompInfo_setInvocationCountAtomic
#0701       1 runFEMacro_invokeExplicitCastHandleConvertArgs
#0702       6 runFEMacro_targetTypeL
#0703       1 runFEMacro_invokeILGenMacrosInvokeExactAndFixup
#0800    5428 CHTable_getAllClassInfo
#0801    5428 CHTable_getClassInfoUpdates
#0900    5734 IProfiler_profilingSample
#0901     632 IProfiler_searchForMethodSample
#1201    3703 KnownObjectTable_getIndexAt
#1203       2 KnownObjectTable_getExistingIndexAt
#1204    2026 KnownObjectTable_symbolReferenceTableCreateKnownObject
```

Reply:
Without implementing the `VM_initializeLocalObjectFlags` message, JITServer sanity functional test might see server crash since this change enabled realTimeGC path that has not been exercised before.

Reply:
If `initializeLocalObjectFlags()` is executed for non-realTimeGC too, then we have a problem today because the following code:
```
   if (!TR::Options::getCmdLineOptions()->realTimeGC())
      {
      initValue |= vmThread()->allocateThreadLocalHeap.objectFlags;
      }
```
accesses `vmThread()->allocateThreadLocalHeap.objectFlags`  for non-realTImeGC case.
On the other hand, if `initializeLocalObjectFlags()` is only entered for realTimeGC, then that `if` statement shown above is never taken.

Reply:
Looked at the code, calling `initializeLocalObjectFlags()` should be independent of realTimeGC. I plan to remove the `VM_initializeLocalObjectFlags ` message. Running a bit more local test first.